frontend/src/components/NavBar.jsx

```diff
@@ -111,6 +111,15 @@ const NavBar = () => {
               </li>
             </ul>
           </div>
+          <button id="docs" className="btn btn-sm btn-secondary border-0 me-1">
```

Comment:
Is `id` attribute necessary?

Reply:
Not necessary, but it's good practice to keep it.

Reply:
Why is it a good practice? I think I've never heard of that. What are the advantages?

Reply:
Although the button text in our code indicates its purpose for documentation, the 'id' attribute can also serve as a form of documentation. It's beneficial for SEO as well, even though we are not currently focusing on that aspect in our case.

I don't see any issues with it, but I'll remove it if you want :)

Reply:
TBH I tried to find something on the web that confirms ids affect SEO, but without much luck - if you have any source on that I'll be happy to have a look.

PS I'm genuinely curious about your motives here and not trying to push anything, so there is no need to remove this id

Reply:
These points were discussed in a course (**Full-Stack Web Development with React Specialization**) offered by Hong Kong University on Coursera. TBH, I didn't cross-verify them as SEO wasn't my primary focus. I didn't place significant emphasis on it and simply included the 'id' attribute based on the information from that course. (considering it as a good practice)😃